rllib/examples/curriculum/curriculum_learning.py

```diff
@@ -148,6 +148,15 @@ def _remote_fn(env_runner, new_task: int):
 class EnvTaskCallback(RLlibCallback):
     """Custom callback implementing `on_train_result()` for changing the envs' maps."""
 
+    def on_algorithm_init(
+        self,
+        *,
+        algorithm: "Algorithm",
+        **kwargs,
+    ) -> None:
+        # Set the initial task to 0.
+        algorithm._counters["current_env_task"] = 0
```

Comment:
Let's not use the `_counters` or `_timers` APIs anymore. Those are very old stack'ish :) 

Instead, do for simple ever increasing counters:
```
algorithm.metrics_logger.log_value("current_env_task", 1, reduce="sum")
```

If the value can change to any other values (even decrease) at any time, do:
```
algorithm.metrics_logger.log_value("[name]", [new, absolute value], window=1)
```